app/molecules/AdminForm/FileUpload.tsx

```diff
@@ -0,0 +1,98 @@
+import { Loader } from '@app/molecules/Loader'
+import { FileType } from '@prisma/client'
+import { Button as SUIButton } from '@singularity/core'
+import { useFormikContext } from 'formik'
+import { useS3Upload } from 'next-s3-upload'
+import { useRef, useState } from 'react'
+import styled from 'styled-components'
+
+type FileUploadProps = {
+  isDisabled?: boolean
+  label: string
+  name: string
+}
+
+// TODO: export in SUI
+const Label = styled.label`
+  font-size: 80%;
+  padding: 0 0 0.25rem 0;
+  font-weight: 500;
+  text-transform: uppercase;
+`
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+`
+
+const Row = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+`
+
+const Button = styled(SUIButton)`
+  width: 200px;
+  height: 42px;
+`
+
+const Thumbnail = styled.img`
+  width: 70px;
+  height: 70px;
+  border-radius: 8px;
+  margin-right: 24px;
+`
+
+const Error = styled.span`
+  margin-top: 8px;
+  font-size: 80%;
+  color: ${p => p.theme.color.danger.default};
+`
+
+export function FileUpload({ isDisabled = false, label, name }: FileUploadProps) {
+  const { errors, isSubmitting, setFieldError, setFieldValue, values } = useFormikContext<any>()
+
+  const inputRef = useRef<HTMLInputElement | null>(null)
+  const { uploadToS3 } = useS3Upload()
+  const [isUploading, setIsUploading] = useState(false)
+
+  const uploadUrl = values?.[name]?.url
+
+  const handleFileChange = async event => {
+    setIsUploading(true)
+    try {
+      const file = event.target.files[0]
+      const { url } = await uploadToS3(file)
+
+      // TODO: Remove useless FileType
+      setFieldValue(name, { title: file.name, type: FileType.EXTERNAL, url })
```

Comment:
For the upload itself in the next PR, I suggest [formidable](https://www.npmjs.com/package/formidable) but you surely already know it. Indeed you won't need to store the file type with that: https://github.com/betagouv/tell-me/blob/main/pages/api/assets/index.ts#L31

Reply:
We actually already have the type, however it doesn't fit the enum you've implemented... And I'm not really sure what's the purpose of keeping the extension stored in the DB, if we have the full URL. WDYT?